core/src/window_service.rs

```diff
@@ -251,7 +251,8 @@ where
                             // with sufficiently large buffers. Needed to ensure
                             // call to `new_from_serialized_shred` is safe.
                             assert_eq!(packet.data.len(), PACKET_DATA_SIZE);
-                            let serialized_shred = packet.data.to_vec();
+                            let mut serialized_shred = packet.data.to_vec();
+                            serialized_shred.truncate(SHRED_PAYLOAD_SIZE);
```

Comment:
Nit: Add a comment here explaining why the truncation is necessary, to remove the nonce and be the proper size for `new_from_serialized_shred`